scripts/ui.py

```diff
@@ -81,15 +85,23 @@ def update_plot(self, plot_obstacles=False):
         # Plot goal
         distance_to_goal = np.linalg.norm(relative_goal)
         if distance_to_goal > 0.1:
-            l = 0.06
-            goal = FancyArrow(l,0,relative_goal[0]-l,relative_goal[1],width=.03, linewidth=0,
+            goal = FancyArrow(lx,0,relative_goal[0]-lx,relative_goal[1],width=.03, linewidth=0,
                                 length_includes_head=True, color="Yellow", alpha=.25, zorder=1)
-            arrow = self.ax.add_patch(goal)
-            self.dynamic_objects.append(arrow)
+            arrow_goal = self.ax.add_patch(goal)
+            self.dynamic_objects.append(arrow_goal)
             self.ax.set_title(f"Distance to goal: {distance_to_goal:.2f} meters", color="gray")
         else:
             self.ax.set_title(" ",color="black")
 
+        # Plot control output
+        if control_output:
+            # Scale QP controller output vector length to same as distance to goal
```

Comment:
Scaling is done to give better view of the controller doing some changes versus the nominal control input. So in general it is equal to "goal arrow" if the QP controller does not change the nominal input. 

Maybe it could be that if the components  `ux, uy `of `control_output` are greater than some threshold value -> plot the arrow.